### PR TITLE
Add Karpenter image.

### PR DESCRIPTION
**Pull request by dlorenc**
Fixes:

Related: 

### Pre-review Checklist

- [ ] **IMPORTANT: 'image-request' tag has been applied if this PR is adding any images, including new versions or variants**

#### For new image PRs only

If you have an apko.yaml file in this PR you need to follow this checklist, otherwise feel free to remove.
- [x] Image is marked experimental or stable as appropriate
- [ ] The last two minor versions are available
- [x] The latest tag points to the newest stable version
- [x] There is a dev tag available that includes a shell and apk tools (by depending on 'wolfi-base')
- [x] The image runs as `nonroot` and GID/UID are set to 65532
  - [ ] Alternatively the username and GID/UID may be a commonly used one from the ecosystem e.g: postgres
  - [ ] See above for exceptions to nonroot rule
- [x] ENTRYPOINT
  - [x] For applications/servers/utilities call main program with no arguments e.g. [redis-server]
  - [ ] For base images leave empty
  - [ ] For dev variants set to entrypoint script that falls back to system
- [x] CMD:
  - [x] For server applications give arguments to start in daemon mode (may be empty)
  - [ ] For utilities/tooling bring up help e.g. `–help`
  - [ ] For base images with a shell, call it e.g. [/bin/sh]
- [x] Consider where and how the image deviates from popular alternatives. Is there a good reason and is it documented?
- [x] Add annotations e.g:
```
annotations:
  "org.opencontainers.image.authors": "Chainguard Team https://www.chainguard.dev/"
  "org.opencontainers.image.url": https://edu.chainguard.dev/chainguard/chainguard-images/reference/busybox/ # use the academy site here
  "org.opencontainers.image.source": https://github.com/chainguard-images/images/tree/main/images/bazel # use github here
```
- [x] Check if environment variables are needed e.g. to set data locations
- [x] Ensure the image responds to SIGTERM
  - [ ] `docker kill $(docker run -d --rm cgr.dev/chainguard/nginx)`
- [x] Documentation. Let's make this excellent. Include usage example.
- [x] Error logs write to stderr and normal logs to stdout. DO NOT write to file.
- [x] Include tests, at the very least a basic smoke test.

### CODE REVIEW

**Comment by jdolitsky** [comment]
needs https://github.com/wolfi-dev/os/pull/863

File: images/karpenter/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+The image is available on `cgr.dev`:
+
+```
+docker pull cgr.dev/chainguard/karpenter:latest
+```
+
+## Using Karpenter
+
+The Chainguard Karpenter image contains the `karpenter` controller and is a drop-in replacement for the upstream image.
```
Comment (imjasonh) [inline images/karpenter/README.md:26]: Looks like you should be able to follow the [official installation instructions](https://karpenter.sh/v0.27.0/getting-started/getting-started-with-eksctl/) and pass this to helm to use our image:

```
--set controller.image=cgr.dev/chainguard/karpenter:latest
```

**Comment by jdolitsky** [comment]
some legitimate build failure here:
```
+ docker run --rm karpenter:7f6869224e3e2aed70ed6169d202d11561255937-latest-amd64
+ grep karpenter
github.com/aws/karpenter-core/pkg/operator.NewOperator()
	/home/build/.cache/go/pkg/mod/github.com/aws/karpenter-core@v0.27.0/pkg/operator/operator.go:71 +0x78
	/home/build/.cache/go/pkg/mod/github.com/aws/karpenter@v0.27.0/cmd/controller/main.go:34 +0x2e
```

**Comment by dlorenc** [comment]
> some legitimate build failure here:
> 
> ```
> + docker run --rm karpenter:7f6869224e3e2aed70ed6169d202d11561255937-latest-amd64
> + grep karpenter
> github.com/aws/karpenter-core/pkg/operator.NewOperator()
> 	/home/build/.cache/go/pkg/mod/github.com/aws/karpenter-core@v0.27.0/pkg/operator/operator.go:71 +0x78
> 	/home/build/.cache/go/pkg/mod/github.com/aws/karpenter@v0.27.0/cmd/controller/main.go:34 +0x2e
> ```

Sorry yeah, it's not a build failure but the image doesn't provide any help or version commands and crashes outside of a cluster. One sec.

**Review by imjasonh — APPROVED** [review]

**Pull request merged**